packages/aws-cdk-lib/custom-resources/README.md

```diff
@@ -824,3 +824,91 @@ new cr.AwsCustomResource(this, 'CrossAccount', {
   })]),
 });
 ```
+
+#### Custom Resource Config
+
+You can set the log retention of every CDK-vended custom resource in a given scope with `CustomResourceConfig`. 
+
+The following example configures every custom resource in this CDK app to retain its logs for ten years:
+```ts
+    import { CustomResourceConfig } from 'aws-cdk-lib/custom-resources';
+
+    CustomResourceConfig.of(app).addLogRetentionLifetime(logs.RetentionDays.TEN_YEARS);
+
+    const stack = new cdk.Stack(app);
+
+    let websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+```
+The following example shows configuring resources of nested stacks to retain its logs for ten years:
+```ts
+export class RootStack extends cdk.Stack {
+    constructor(scope: Construct) {
+        super(scope, "rootStack");
+        new AlphaStack(this);
+        new BetaStack(this);
+        CustomResourceConfig.of(this).addLogRetentionLifetime(logs.RetentionDays.TEN_YEARS);
+    }
+}
+export class AlphaStack extends cdk.NestedStack {
+    constructor(scope: Construct) {
+        super(scope, "AlphaStack");
+        let websiteBucket = new s3.Bucket(this, "WebsiteBucket", {});
+        new s3deploy.BucketDeployment(this, "s3deployNone", {
+            sources: [s3deploy.Source.jsonData("file.json", { a: "b" })],
+            destinationBucket: websiteBucket,
+            logRetention: logs.RetentionDays.ONE_DAY,
+        });
+    }
+}
+export class BetaStack extends cdk.NestedStack {
+    constructor(scope: Construct) {
+        super(scope, "BetaStack");
+        let websiteBucket = new s3.Bucket(this, "WebsiteBucket", {});
+        new s3deploy.BucketDeployment(this, "s3deployNone", {
+            sources: [s3deploy.Source.jsonData("file.json", { a: "b" })],
+            destinationBucket: websiteBucket,
+            logRetention: logs.RetentionDays.ONE_DAY,
+        });
+    }
+}
+```
+
+The following example shows configuring resources of two top-level stacks to retain its logs for ten years:
+```tsc
+export class RootStack extends cdk.Stack {
+    constructor(scope: Construct) {
+        super(scope, "rootStack");
+        new AlphaStack(this);
+        new BetaStack(this);
+        CustomResourceConfig.of(this).addLogRetentionLifetime(
+            logs.RetentionDays.TEN_YEARS
+        );
+    }
+}
+export class AlphaStack extends cdk.Stack {
```

Comment:
`AlphaStack` and `BetaStack` are not top-level stacks because they're in the scope of `RootStack`. To be top-level, they have to be direct children of an `App`.